logic/src/commonMain/kotlin/com/wire/kalium/logic/data/connection/ConnectionRepository.kt

```diff
@@ -184,34 +201,33 @@ internal class ConnectionDataSource(
                 userDAO.getUserByQualifiedID(selfUserID)
                     .filterNotNull()
                     .map(userMapper::fromDaoModelToSelfUser)
-            }.firstOrNull() ?: throw  IllegalStateException()
-    }
-
-    private suspend fun updateUserConnectionStatus(connections: List<ConnectionDTO>) {
-        connections.forEach { connection ->
-            when (connection.status) {
-                ConnectionStateDTO.PENDING -> updateConversationMemberFromConnection(connection)
-                ConnectionStateDTO.SENT -> updateConversationMemberFromConnection(connection)
-                ConnectionStateDTO.BLOCKED -> updateConversationMemberFromConnection(connection)
-                ConnectionStateDTO.IGNORED -> updateConversationMemberFromConnection(connection)
-                ConnectionStateDTO.CANCELLED -> {
-                    kaliumLogger.d("skipping cancelled request")
-                }
-                ConnectionStateDTO.MISSING_LEGALHOLD_CONSENT -> updateConversationMemberFromConnection(connection)
-                ConnectionStateDTO.ACCEPTED -> updateConversationMemberFromConnection(connection)
-            }
-        }
+            }.firstOrNull() ?: throw IllegalStateException()
     }
 
-    private suspend fun updateConversationMemberFromConnection(connectionDTO: ConnectionDTO) {
+    private suspend fun updateConversationMemberFromConnection(connection: Connection) =
         wrapStorageRequest {
             conversationDAO.updateOrInsertOneOnOneMemberWithConnectionStatus(
-                userId = idMapper.fromApiToDao(connectionDTO.qualifiedToId),
-                status = connectionStatusMapper.fromApiToDao(state = connectionDTO.status),
-                conversationID = idMapper.fromApiToDao(connectionDTO.qualifiedConversationId)
+                userId = idMapper.toDaoModel(connection.qualifiedToId),
+                status = connectionStatusMapper.toDaoModel(connection.status),
+                conversationID = idMapper.toDaoModel(connection.qualifiedConversationId)
             )
         }.onFailure {
-            kaliumLogger.e("There was an error when trying to persist the connection: $connectionDTO")
+            kaliumLogger.e("There was an error when trying to persist the connection: $connection")
+        }
+
+    /**
+     * This will update the connection status on user table and will insert members only
+     * if the [ConnectionDTO.status] is other than [ConnectionStateDTO.PENDING] or [ConnectionStateDTO.SENT]
+     */
+    private suspend fun handleUserConnectionStatusPersistence(connection: Connection): Either<CoreFailure, Unit> =
+        when (connection.status) {
+            ConnectionState.NOT_CONNECTED,
+            ConnectionState.PENDING,
```

Comment:
is there a reason why `PENDING` and `NOT_CONNECTED` are ignored ?, maybe add some comments about the reason.